python/lsst/daf/butler/_butler.py

```diff
@@ -798,6 +799,83 @@ def get_dataset_type(self, name: str) -> DatasetType:
         """
         raise NotImplementedError()
 
+    @abstractmethod
+    def find_dataset(
+        self,
+        datasetType: DatasetType | str,
+        dataId: DataId | None = None,
+        *,
+        collections: CollectionArgType | None = None,
+        timespan: Timespan | None = None,
+        datastore_records: bool = False,
```

Comment:
I had completely failed to notice that that flag was in there now. One option is for `get_dataset` to always expand and add everything. People shouldn't be calling it in bulk.

Reply:
Presumably get_dataset should have the datastore_records and dimension_records booleans as well for consistency?

Reply:
:+1:

Reply:
I also need to add datastore records to the SerializedDatasetRef because that was never updated and so it's impossible to return datastore records from the server at the moment...

Reply:
Jim asked me to not add them to `SerializedDatasetRef`, as it may increase graph size on disk (but I don't think we give records to refs that go into the graph, at least not yet).

Reply:
We are sending SerializedDatasetRef over the wire so we can only reconstruct things in the far end that were put in at the other end. We need to add them somehow (but obviously they should be optional). For now I've turned the feature off in client/server so you can't get the datastore records.